prototype/simulator/container/src/destination/destination.js

```diff
@@ -44,9 +46,71 @@ class Destination {
 		this.cognitoUser = usr
 		this.userData.creds = creds
 		this.userData.clientId = `${this.userData.identity}:` + Date.now()
-		this.orderInterval = setInterval(
-			this.sendOrder.bind(this),
-			utils.getMsFromParams(this.params))
+
+		// if event file is not present would generate orders based on the time interval
+		if (!this.params.eventsFilePath) {
+			this.orderInterval = setInterval(
+				this.sendOrder.bind(this),
+				utils.getMsFromParams(this.params),
+			)
+		} else {
+			// otherwise replay the events after waiting 2 seconds
+			setTimeout(
+				this.replayEvents.bind(this),
+				2000,
+			)
+		}
+	}
+
+	async replayEvents () {
+		logger.info('Replaying event file')
+
+		const nowTime = dayjs().format('HHmmss') - 2 // seconds tolerance for the preparation phase
+		const events = await helper.loadRemoteFile(this.config.simulatorConfigBucket, this.params.eventsFilePath)
+		const validOrders = events.orders.filter(q => {
+			const bookedTime = dayjs(q.payload.bookedAt).format('HHmmss')
+
+			return bookedTime > nowTime
+		}).sort((a, b) => {
+			return dayjs(a.payload.bookedAt).format('HHmmssSSS') - dayjs(b.payload.bookedAt).format('HHmmssSSS')
+		}).reduce((acc, curr) => {
+			const bookedTmeSec = dayjs(curr.payload.bookedAt).format('HHmmss')
+
+			if (!acc[bookedTmeSec]) {
+				acc[bookedTmeSec] = [curr]
+			} else {
+				acc[bookedTmeSec].push(curr)
+			}
+
+			return acc
+		}, {})
+
+		logger.debug('Original Events: ', events)
```

Comment:
let's log only `.length`. if we get 10s of thousands of events per day this won't be performant